### ISSUE TITLE
MacBook stuck on "barrier is starting" and can not connect to windows PC.

### ISSUE THREAD

**Issue by chelan-athlanius** [debauchee/barrier#1154]
**Describe the bug**
When I try to connect my Mac as a client to my Windows 10 PC, it gets stuck on "barrier is starting" and will not connect. It works perfectly fine the other way around when my mac is the server, but that isn't helpful.
<img width="838" alt="Screen Shot 2021-05-02 at 2 21 23 PM" src="https://user-images.githubusercontent.com/83434049/116824826-dfc71900-ab51-11eb-9e0e-bc9f0c0dbf54.png">

**Expected behavior**
Mac connects to PC.

**Desktop (please complete the following information):**

 - OS: Mac to Windows
- Barrier version [e.g 2.3.3]

**Comment by mtuthill** [comment]
I also get this issue

**Comment by hdwatts** [comment]
This just started happening to me, I believe I had an image in my clipboard that was somehow screwing it up (logs below).

```
[2021-05-07T15:06:20] INFO: leaving screen
[2021-05-07T15:06:20] INFO: bitmap: 1046x-1096 32
[2021-05-07T15:06:20] INFO: convert image from: depth=32 comp=3
[2021-05-07T15:06:20] FATAL: An error occurred: string too long
[2021-05-07T15:06:21] ERROR: failed to launch, error: process immediately stopped
[2021-05-07T15:06:21] INFO: backing off, wait=4s, failures=2
[2021-05-07T15:06:25] INFO: starting new process as privileged user
[2021-05-07T15:06:25] ERROR: failed to get desktop path, no drop target available, error=2
```

I spammed CTRL+C on some random text on my windows machine a few times to fully reset the clipboard, then restarted the server on my mac, and the server now started successfully.

**Comment by chelan-athlanius** [comment]
Didn't work.

**Comment by stipid** [comment]
try to restart win10 computer.

**Comment by dianjiao03** [comment]
Same issue here, and none of the solutions seem to work.

**Comment by mhutchins** [comment]
Same issue - however, restarting the windows PC (master) solved the issue for me.

**Comment by NuclearMonster** [comment]
> I spammed CTRL+C on some random text on my windows machine a few times to fully reset the clipboard, then restarted the server on my mac, and the server then started successfully.
I had this same issue, thanks for the workaround!

 My console log was a little different, there was a hint:
`[2021-08-31T18:37:59] FATAL: An error occurred: string too long`

**Comment by reikolydia** [comment]
Have the same issue, clearing clipboard on server sure did help.

The connections were initially successful, but i cannot move the mouse over to another client from the server as it would simply attempt to restart the process from both client and server and give that "string too long" error. i did not restart any client/server after clearing the clipboard however.

**Comment by wesgurn** [comment]
Can confirm the workaround of clearing the clipboard or copying some small text on the server side restores the client connection and fixes issue.

**Comment by juanmartin** [comment]
In my case clearing Barrier on the server (Windows) logs helped 🤷 
My client is mac.

**Comment by radieon** [comment]
Copying some short text to my server's clipboard allowed me to connect to my clients after receiving this same error.

**Comment by zhangineer2** [comment]
copying short text to my server's clipboard solved the problem for me as well

**Comment by jandestiny** [comment]
Same solution for me: Copying a short text to the clipboard got barrier out of the "string too long" error loop. Weird bug.

**Comment by pkxpp** [comment]
> This just started happening to me, I believe I had an image in my clipboard that was somehow screwing it up (logs below).
> 
> ```
> [2021-05-07T15:06:20] INFO: leaving screen
> [2021-05-07T15:06:20] INFO: bitmap: 1046x-1096 32
> [2021-05-07T15:06:20] INFO: convert image from: depth=32 comp=3
> [2021-05-07T15:06:20] FATAL: An error occurred: string too long
> [2021-05-07T15:06:21] ERROR: failed to launch, error: process immediately stopped
> [2021-05-07T15:06:21] INFO: backing off, wait=4s, failures=2
> [2021-05-07T15:06:25] INFO: starting new process as privileged user
> [2021-05-07T15:06:25] ERROR: failed to get desktop path, no drop target available, error=2
> ```
> 
> I spammed CTRL+C on some random text on my windows machine a few times to fully reset the clipboard, then restarted the server on my mac, and the server then started successfully.

It works!Thanks so much.

**Comment by smenendez19** [comment]
> This just started happening to me, I believe I had an image in my clipboard that was somehow screwing it up (logs below).
> 
> ```
> [2021-05-07T15:06:20] INFO: leaving screen
> [2021-05-07T15:06:20] INFO: bitmap: 1046x-1096 32
> [2021-05-07T15:06:20] INFO: convert image from: depth=32 comp=3
> [2021-05-07T15:06:20] FATAL: An error occurred: string too long
> [2021-05-07T15:06:21] ERROR: failed to launch, error: process immediately stopped
> [2021-05-07T15:06:21] INFO: backing off, wait=4s, failures=2
> [2021-05-07T15:06:25] INFO: starting new process as privileged user
> [2021-05-07T15:06:25] ERROR: failed to get desktop path, no drop target available, error=2
> ```
> 
> I spammed CTRL+C on some random text on my windows machine a few times to fully reset the clipboard, then restarted the server on my mac, and the server then started successfully.

This works, thanks!

**Comment by FurmanSK** [comment]
Yeah just came across this error. Saw it was converting an image and string too long error. Can we get an option to disable images in clipboard to be ignored? It doesn't even work anyways for me going from win10(server) to mac m2(client).
